### PR TITLE
For #20927: Replace icons with AC icons.

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/perf/ThreadPenaltyDeathWithIgnoresListener.kt [hunk 1]
```diff
@@ -35,7 +35,7 @@ class ThreadPenaltyDeathWithIgnoresListener(
         if (shouldViolationBeIgnored(violation)) {
             logger.debug("Ignoring StrictMode ThreadPolicy violation", violation)
         } else {
-            penaltyDeath(violation)
+            // penaltyDeath(violation)
```
Comment (MozillaNoah) [inline app/src/main/java/org/mozilla/fenix/perf/ThreadPenaltyDeathWithIgnoresListener.kt:38]: is this a temporary comment? should a comment be added to give some context why this was taken out?

File: app/src/main/java/org/mozilla/fenix/search/awesomebar/ShortcutsSuggestionProvider.kt [hunk 2]
```diff
@@ -26,7 +29,12 @@ class ShortcutsSuggestionProvider(
     override val id: String = UUID.randomUUID().toString()
 
     private val settingsIcon by lazy {
-        AppCompatResources.getDrawable(context, R.drawable.ic_settings)?.toBitmap()
+        AppCompatResources.getDrawable(context, R.drawable.mozac_ic_settings)?.apply {
```
Comment (MozillaNoah) [inline app/src/main/java/org/mozilla/fenix/search/awesomebar/ShortcutsSuggestionProvider.kt:32]: how come this is being colored at runtime and not in the vector xml?

File: app/src/main/java/org/mozilla/fenix/perf/ThreadPenaltyDeathWithIgnoresListener.kt (hunk 1 above)
Comment (mcarare) [reply app/src/main/java/org/mozilla/fenix/perf/ThreadPenaltyDeathWithIgnoresListener.kt:38]: It's a local change for one of my test phones. I will remove it.

File: app/src/main/java/org/mozilla/fenix/search/awesomebar/ShortcutsSuggestionProvider.kt (hunk 2 above)
Comment (mcarare) [reply app/src/main/java/org/mozilla/fenix/search/awesomebar/ShortcutsSuggestionProvider.kt:32]: The vector in AC has other colors (`mozac_ui_icons_fill`). Most of the components in AC are using also a tint besides the image resource, a tint that we pass from the client (see `BrowserMenuImageText` and other `BrowserMenuItem` menu items). In places where we use the resource directly, we have to tint them ourselves. (The overriding of `mozac_ui_icons_fill` proved to have other downsides due to having  the private theme.)

File: app/src/main/java/org/mozilla/fenix/perf/ThreadPenaltyDeathWithIgnoresListener.kt (hunk 1 above)
Comment (pocmo) [reply app/src/main/java/org/mozilla/fenix/perf/ThreadPenaltyDeathWithIgnoresListener.kt]: Heh, I do that locally all the time too. We need to fix this thing. 😅

**Review by MozillaNoah — APPROVED** [review]

**Pull request merged**